dags/nyc_taxi_dag.py

```diff
@@ -1,138 +1,65 @@
         PostgresOperator(
-            task_id=f"log_{file_name}",
-            postgres_conn_id="cratedb_demo_connection",
-            sql=Path("include/taxi-insert.sql").read_text(encoding="utf-8"),
-        ).execute({})
+                task_id="copy_new_csv_file",
+                postgres_conn_id="cratedb_demo_connection",
+                sql=f"""
+                        COPY nyc_taxi.load_trips_staging
+                        FROM '{new_csv_file}'
+                        WITH (format = 'csv', empty_string_as_null = true)
+                        RETURN SUMMARY;
+                    """
+            ).execute({})
 
         PostgresOperator(
-            task_id=f"mark_processed_{file_name}",
+            task_id="log_new_csv_file",
             postgres_conn_id="cratedb_demo_connection",
-            sql=f"INSERT INTO nyc_taxi.load_files_processed VALUES ('{missing_url}');",
+            sql=Path('include/taxi-insert.sql').read_text(encoding="utf-8"),
         ).execute({})
 
         PostgresOperator(
-            task_id=f"purge_staging_{file_name}",
+            task_id="purge_staging_new_csv_file",
             postgres_conn_id="cratedb_demo_connection",
-            sql="DELETE FROM nyc_taxi.load_trips_staging;",
+            sql="DELETE FROM nyc_taxi.load_trips_staging;"
         ).execute({})
 
+    process_new_files(download_from_s3_csv(get_month_year()))
 
-with DAG(
-    dag_id="nyc-taxi",
-    start_date=pendulum.datetime(2021, 11, 11, tz="UTC"),
-    schedule_interval="@daily",
-    catchup=False,
-) as dag:
-    download_data_urls = SimpleHttpOperator(
-        task_id="download_data_urls",
-        method="GET",
-        http_conn_id="http_raw_github",
-        endpoint="toddwschneider/nyc-taxi-data/master/setup_files/raw_data_urls.txt",
-        headers={},
-    )
-
-    clean_data_urls = PythonOperator(
-        task_id="clean_data_urls",
-        python_callable=clean_data_urls,
-        op_kwargs={},
-    )
-
-    clean_data_urls << download_data_urls
-
-    get_processed_files = PythonOperator(
-        task_id="get_processed_files",
-        python_callable=get_processed_files,
-        op_kwargs={},
-    )
-
-    identitfy_missing_urls = PythonOperator(
-        task_id="identitfy_missing_urls",
-        python_callable=identitfy_missing_urls,
-        op_kwargs={},
-    )
-
-    identitfy_missing_urls << [clean_data_urls, get_processed_files]
-
-    # The staging table should be empty already. Purging it again in case of
-    # an abort or other error case.
-    purge_staging_init = PostgresOperator(
-        task_id="purge_staging_init",
-        postgres_conn_id="cratedb_demo_connection",
-        sql="DELETE FROM nyc_taxi.load_trips_staging;",
-    )
-
-    process_new_files = PythonOperator(
-        task_id="process_new_files",
-        python_callable=process_new_files,
-        op_kwargs={},
-    )
-
-    process_new_files << [identitfy_missing_urls, purge_staging_init]
+dag = taskflow()
```

Comment:
```suggestion
taskflow()
```
After rebasing, with Airflow 2.4, assignment to `dag` isn't needed anymore.